### PR TITLE
Get rid of libsoup2

### PR DESCRIPTION
**Pull request by twolife**
Compile-tested only. Since commit 39a47f6aa1780fb1fcb6f95130802acadbdfbe59 this code doesn't seems to have any real use.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -317,14 +317,14 @@ list(APPEND LIBS ${LIBXML2_LIBRARIES})
 add_definitions(${LIBXML2_DEFINITIONS})
 
 # used to launch a tiny http server, which could be used by OAuth services
-find_package(LibSoup2)
-if(LibSoup2_FOUND)
-  include_directories(SYSTEM ${LibSoup2_INCLUDE_DIRS})
-  list(APPEND LIBS ${LibSoup2_LIBRARIES})
-  add_definitions(${LibSoup2_DEFINITIONS})
+pkg_check_modules(LIBSOUP3 libsoup-3.0)
```
Comment (kmilos) [inline src/CMakeLists.txt:320]: If there is no FindLibSoup3.cmake shipped w/ CMake (or libsoup3 doesn't ship CMake config files), we should create one. Don't like the idea of using pkg_* macros directly in our main CMake recipe...

**Comment by kmilos** [comment]
Not sure this is a good time for the switch, as Debian 11 and Ubuntu 20.04 are still not on v3...

**Comment by parafin** [comment]
If there's no real use, why keep it and maintain? Just remove it.

**Comment by twolife** [comment]
> If there's no real use, why keep it and maintain? Just remove it.

[done]

**Comment by kmilos** [comment]
There's quite a few more places to remove it from (CI, build instructions, ...)

File: .ci/ci-script-windows.sh [hunk 2]
```diff
@@ -82,7 +82,7 @@ cat > "$FONTCONFIG_FILE" <<EOF
 EOF
 
 execute 'Installing base-devel and toolchain'  pacman -S --needed --noconfirm mingw-w64-x86_64-{toolchain,clang,cmake}
-execute 'Installing dependencies' pacman -S --needed --noconfirm  mingw-w64-x86_64-{exiv2,lcms2,dbus-glib,openexr,sqlite3,libxslt,libsoup,libavif,libwebp,libsecret,lua,graphicsmagick,openjpeg2,gtk3,pugixml,libexif,osm-gps-map,libgphoto2,flickcurl,drmingw,gettext,python3,iso-codes,lensfun}
+execute 'Installing dependencies' pacman -S --needed --noconfirm  mingw-w64-x86_64-{exiv2,lcms2,dbus-glib,openexr,sqlite3,libxslt,libavif,libwebp,libsecret,lua,graphicsmagick,openjpeg2,gtk3,pugixml,libexif,osm-gps-map,libgphoto2,flickcurl,drmingw,gettext,python3,iso-codes,lensfun}
```
Comment (kmilos) [inline .ci/ci-script-windows.sh:85]: Thanks! Btw, you can leave out this particular file from the commit - it is currently unused, and it's planned for removal anyway....

**Review by TurboGit — APPROVED** [review]
I'll merge the MacOS issue is a CI bug now corrected. So should be fine. Thanks for the clean-up.

**Pull request merged**